src/mqtt-broker/src/subscribe/sub_common.rs

```diff
@@ -75,21 +75,32 @@ pub fn sub_path_validator(sub_path: String) -> bool {
 
 pub fn path_regex_match(topic_name: String, sub_path: String) -> bool {
     let path = if is_share_sub(sub_path.clone()) {
-        let (_, group_path) = decode_share_info(sub_path);
+        let (_, group_path) = decode_share_info(sub_path.clone());
         group_path
+    } else if is_queue_sub(sub_path.clone()) {
+        decode_queue_info(sub_path.clone())
     } else {
-        sub_path
+        sub_path.clone()
+    };
+
+    let topic = if is_share_sub(topic_name.clone()) {
```

Comment:
I can't follow the logic of this code. topic has no concept of shared subscriptions